libcaf_test/caf/test/runner.cpp

```diff
@@ -13,23 +13,64 @@
 #include "caf/config_option_adder.hpp"
 #include "caf/config_option_set.hpp"
 #include "caf/detail/log_level.hpp"
+#include "caf/detail/set_thread_name.hpp"
 
+#include <condition_variable>
 #include <iostream>
+#include <mutex>
 #include <optional>
 #include <regex>
 #include <string>
+#include <thread>
 
 namespace caf::test {
 
 namespace {
 
+class watchdog {
+public:
+  watchdog(int secs) {
+    using detail::format_to;
+    thread_ = secs <= 0 ? std::thread() : std::thread{[=] {
+      caf::detail::set_thread_name("test.watchdog");
+      auto tp = std::chrono::high_resolution_clock::now()
+                + std::chrono::seconds(secs);
+      std::unique_lock<std::mutex> guard{mtx_};
+      while (!canceled_
+             && cv_.wait_until(guard, tp) != std::cv_status::timeout) {
+      }
+      if (!canceled_) {
+        auto err_out = [] { return std::ostream_iterator<char>{std::cerr}; };
+        format_to(err_out(),
+                  "WATCHDOG: unit test did not finish within {} seconds\n",
+                  secs);
+        abort();
+      }
+    }};
+  }
+  ~watchdog() {
+    {
+      std::lock_guard<std::mutex> guard{mtx_};
+      canceled_ = true;
+      cv_.notify_all();
+    }
+    if (thread_.joinable())
+      thread_.join();
+  }
+
+  bool canceled_ = false;
+  std::mutex mtx_;
+  std::condition_variable cv_;
+  std::thread thread_;
+};
+
 config_option_set make_option_set() {
   config_option_set result;
   config_option_adder{result, "global"}
     .add<bool>("available-suites,a", "print all available suites")
     .add<bool>("help,h?", "print this help text")
     .add<bool>("no-colors,n", "disable coloring (ignored on Windows)")
-    .add<size_t>("max-runtime,r", "set a maximum runtime in seconds")
+    .add<size_t>("max-runtime,m", "set a maximum runtime in seconds")
```

Comment:
I agree that `m` is the more logical choice. But please leave it at `r` for backwards compatibility with the old framework. We will have them both side-by-side for a while, so we should be careful about breaking the CLI.

Reply:
Ah, I know why you've changed it. I had `-m` in the ticket. Sorry about that, my mistake (I've updated the description in the ticket).